src/detect-filesize.c

```diff
@@ -257,36 +249,28 @@ static int DetectFilesizeInitTest(
         DetectEngineCtx **de_ctx, Signature **sig, DetectU64Data **fsd, const char *str)
 {
     char fullstr[1024];
-    int result = 0;
 
     *de_ctx = NULL;
     *sig = NULL;
 
-    if (snprintf(fullstr, 1024, "alert http any any -> any any (msg:\"Filesize "
-                                "test\"; filesize:%s; sid:1;)", str) >= 1024) {
-        goto end;
-    }
+    FAIL_IF(snprintf(fullstr, 1024,
+                    "alert http any any -> any any (msg:\"Filesize "
+                    "test\"; filesize:%s; sid:1;)",
+                    str) >= 1024);
 
     *de_ctx = DetectEngineCtxInit();
-    if (*de_ctx == NULL) {
-        goto end;
-    }
+    FAIL_IF_NULL(*de_ctx);
 
     (*de_ctx)->flags |= DE_QUIET;
 
     (*de_ctx)->sig_list = SigInit(*de_ctx, fullstr);
```

Comment:
could you switch this to use `DetectEngineAppendSig`?

Reply:
Would it be like this:
` FAIL_IF_NULL(DetectEngineAppendSig(de_ctx,"alert http any any -> any any (msg:\"Filesize " "test\"; filesize:%s; sid:1;)");`

Reply:
Can't it be `FAIL_IF_NULL(DetectEngineAppendSig(de_ctx, fullstr)`?